### ISSUE TITLE
Motoko 0.11 `<system>` not working for class bodies?

### ISSUE THREAD

**Issue by Gekctek** [dfinity/motoko#4447]
From what I understand the classes should just put a <system> type as the first type and calling system functions should work.
But I have this code and 
```motoko
public class Dao<system, TProposalContent>(
        data : StableData<TProposalContent>,
        onExecute : Proposal<TProposalContent> -> async* (),
        onReject : Proposal<TProposalContent> -> async* (),
    ) {
     // ....

        public func resetEndTimers<system>() {
            for (proposal in proposals.vals()) {
                switch (proposal.endTimerId) {
                    case (null) ();
                    case (?id) Timer.cancelTimer(id);
                };
                proposal.endTimerId := null;
                if (proposal.status == #open) {
                    let proposalDurationNanoseconds = durationToNanoseconds(proposalDuration);
                    let endTimerId = createEndTimer<system>(proposal.id, proposalDurationNanoseconds);
                    proposal.endTimerId := ?endTimerId;
                };
            };
        };
// ...
        resetEndTimers<system>(); // Always reset end timers on init
 ````
It gives me an error with resetEndTimers about `system` capability required, but not available
![image](https://github.com/dfinity/motoko/assets/1934953/f19d293e-be97-4058-9b59-73c4e1f5215b)

Am I doing this right? It seems from the docs that classes are allowed

**Comment by crusso** [comment]
Yes, I think that's a bug. Thanks for reporting it!

A temporary workaround might be to define the class as a type definition, without the system parameter, and a function with the parameter returning an object.  But I haven't actually tried that.

**Comment by ggreif** [comment]
Right, I believe this currently only works with `actor`. Thanks for the report!

**Comment by ggreif** [comment]
@Gekctek try to prepend `ignore` to the problematic line as a workaround. This puts the type checker into _checking mode_ (as opposed to _inferring mode_), and that may be the crucial difference.

**Comment by ggreif** [comment]
@Gekctek 0.11.1 will have the fix of #4453, and then you can remove the workaround I suggested above.
